src/types.rs

```diff
@@ -117,6 +117,15 @@ fn long_signature(name: &str, params: &[ParamType]) -> Hash {
     H256::from(result)
 }
 
+/// Validate Etherium address from string and return EthAddress
+pub fn validate_eth_address(address: String) -> EthAddress {
+    let data = hex::decode(address).expect("ETH_ADDRESS_FAILED");
+    assert_eq!(data.len(), 20, "ETH_WRONG_ADDRESS_LENGTH");
+    let mut result = [0u8; 20];
+    result.copy_from_slice(&data);
+    result
+}
+
```

Comment:
As we are changing this function in the PR and we are striving to have a more testable environment, I'd suggest returning an error instead of panicking from the function.